### PR TITLE
Add documentation on using RAJA in other projects

### PR DESCRIPTION
**Pull request by davidbeckingsale**
# Summary

- This PR is a documentation improvement
- It does the following:
  - Adds documentation on using RAJA in other projects

### CODE REVIEW

File: docs/sphinx/user_guide/using_raja.rst [hunk 1]
```diff
@@ -18,6 +18,20 @@ are visible, and linking against the RAJA library. We maintain a
 shows how to use RAJA in a CMake project, either as a Git submodule or
 as an externally installed library that you link your application against.
 
+It is important to ensure that you pass in compiler options in your project
+that match the options that you compiled RAJA with. Specifically, if you built
+RAJA with `-DENABLE_OPENMP=On` then you need to add a flag like `-fopenmp` when
```
Comment (rhornung67) [inline docs/sphinx/user_guide/using_raja.rst]: I would change this to 'you need to pass to the compiler the OpenMP flag appropriate for the compiler, such as `-fopenmp`.

File: docs/sphinx/user_guide/using_raja.rst [hunk 2]
```diff
@@ -18,6 +18,20 @@ are visible, and linking against the RAJA library. We maintain a
...
+
+Similarly, if you built RAJA with `-DENABLE_CUDA=On` then you need to pass in
+some extra compile flags, and also ensure you use `nvcc` to compile any files
+in your project that include RAJA. The required flags are:
+
+- `-std=c++11`
+- `-x cu ` required if your file extensions are not .cu
+- `--expt-extended-lambda` 
+- `-arch=smXX` where `smXX` is one of: 
+
```
Comment (rhornung67) [inline docs/sphinx/user_guide/using_raja.rst]: Is this incomplete? 'one of'....

**Comment by rhornung67** [comment]
Should we add a link to the template project in the user guide?

File: docs/sphinx/user_guide/using_raja.rst (hunk 2 above)
Comment (davidbeckingsale) [reply docs/sphinx/user_guide/using_raja.rst]: Yup, need to put in the list.

**Comment by davidbeckingsale** [comment]
This PR is currently a draft. @rhornung67 - we do link to the template project.

**Comment by rhornung67** [comment]
@davidbeckingsale what is the status of this PR?

**Comment by davidbeckingsale** [comment]
@rhornung67 - needs finishing up still.

**Comment by rhornung67** [comment]
OK. Thanks. Please let me know when you want a review.

**Pull request closed without merging**